pkg/controller/storage/capability/capability_controller.go

```diff
@@ -126,24 +116,10 @@ func NewController(
 		DeleteFunc: controller.enqueueStorageClass,
 	})
 
-	// ProvisionerCapability acts as a value source of its relevant StorageClassCapabilities
-	// so when a PC is created/updated, the corresponding SCCs should be created(if not exists)/updated
-	// we achieve this by simply enqueueing the StorageClasses of the same provisioner
-	// but don't overdo by cascade deleting the SCCs when a PC is deleted
-	// since the role of PCs is more like a template rather than owner to SCCs
-
-	// This is a backward compatible fix to remove the useless auto detection of SCCs
-	// in the future, we will only keep ProvisionerCapability and remove the StorageClassCapability CRD entirely
-	ksStorageInformer.ProvisionerCapabilities().Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc: controller.handleProvisionerCapability,
-		UpdateFunc: func(oldObj, newObj interface{}) {
-			newPC := newObj.(*ksstorage.ProvisionerCapability)
-			oldPC := oldObj.(*ksstorage.ProvisionerCapability)
-			if newPC.ResourceVersion == oldPC.ResourceVersion {
-				return
-			}
-			controller.handleProvisionerCapability(newObj)
-		},
+	csiDriverInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    controller.enqueueStorageClassByCSI,
+		UpdateFunc: nil,
+		DeleteFunc: controller.enqueueStorageClassByCSI,
```

Comment:
do we need handle the delete event? What would you do if a csi driver was deleted? Remove the allow-xxx annotations?

Reply:
yes，the annotations will be removed.

Reply:
I did not see the remove annotation code. In my opinion, the annotation should not change even if the csi driver is deleted. just like the `allowExpansion`.  They are independent. `I allow it, but the driver does not exist`.

Reply:
> I did not see the remove annotation code. In my opinion, the annotation should not change even if the csi driver is deleted. just like the `allowExpansion`. They are independent. `I allow it, but the driver does not exist`.

Okay